source/scheduler.cpp

```diff
@@ -0,0 +1,31 @@
+#include "scheduler.hpp"
+
+#include <spdlog/spdlog.h>
+#include <iostream>
+#include <queue>
+
+#include "event.hpp"
+
+namespace sim {
+
+void Scheduler::tick() {
```

Comment:
I think `tick()` should return boolean value to notify simulator when `m_events` is empty.

```suggestion
bool Scheduler::tick() {
```

Reply:
Can you provide some example when we need such behaviour?

Reply:
```